pkg/api/service/scans.go

```diff
@@ -136,38 +68,49 @@ type scanStats struct {
 	NumberOfChecksPerChecktype map[string]int
 }
 
+// ChecktypesByAssettypes is used as a lookup table to check if a checktype can
+// be run against a concrete assettype.
+type ChecktypesByAssettypes map[string]map[string]struct{}
+
 // ScansService implements the functionality needed to create and query scans.
 type ScansService struct {
-	db            ScansPersistence
-	logger        log.Logger
-	vulcanCore    CoreAPI
-	notifier      ScanNotifier
-	ccreator      *CheckCreator
-	metricsClient metrics.Client
-	accreator     AsyncChecksCreator
+	db             persistence.ScansStore
+	logger         log.Logger
+	ctInformer     ChecktypesInformer
+	metricsClient  metrics.Client
+	ccreator       ChecksCreator
+	scansNotifier  notify.Notifier
```

Comment:
Same question than the one related to the persistence interface above.

Reply:
You mean for the `Notifier` interface?
Yes, similar answer to above, for theses cases where it's already a small interface or all "originally exposed methods" are used by the component that instantiates that interface, I think it's not worth to create more abstraction layers in between that end up adding complexity to the code.

Reply:
I agree makes sense in this case.